unified-runtime/source/adapters/opencl/device.cpp

```diff
@@ -1418,6 +1418,63 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
     return ReturnValue(true);
   case UR_DEVICE_INFO_KERNEL_LAUNCH_CAPABILITIES:
     return ReturnValue(0);
+  case UR_DEVICE_INFO_LUID: {
+    // LUID is only available on Windows.
+    // Intel extension for device LUID. This returns the LUID as
+    // std::array<std::byte, 8>. For details about this extension,
+    // see sycl/doc/extensions/supported/sycl_ext_intel_device_info.md.
+
+    // Use the cl_khr_device_uuid extension, if available.
+    bool isKhrDeviceLuidSupported = false;
+    if (hDevice->checkDeviceExtensions({"cl_khr_device_uuid"},
+                                       isKhrDeviceLuidSupported) !=
+            UR_RESULT_SUCCESS ||
+        !isKhrDeviceLuidSupported) {
+      return UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION;
+    }
+
+    cl_bool isLuidValid;
+    CL_RETURN_ON_FAILURE(
+        clGetDeviceInfo(hDevice->CLDevice, CL_DEVICE_LUID_VALID_KHR,
+                        sizeof(cl_bool), &isLuidValid, nullptr));
+
+    if (!isLuidValid) {
+      return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+    }
+
+    static_assert(CL_LUID_SIZE_KHR == 8);
+    std::array<unsigned char, CL_LUID_SIZE_KHR> UUID{};
+    CL_RETURN_ON_FAILURE(clGetDeviceInfo(hDevice->CLDevice, CL_DEVICE_LUID_KHR,
+                                         UUID.size(), UUID.data(), nullptr));
+    return ReturnValue(UUID);
+  }
+  case UR_DEVICE_INFO_NODE_MASK: {
+    // Device node mask is only available on Windows.
+    // Intel extension for device node mask. This returns the node mask as
+    // uint32_t. For details about this extension,
+    // see sycl/doc/extensions/supported/sycl_ext_intel_device_info.md.
+
+    // Use the cl_khr_device_uuid extension, if available.
+    bool isKhrDeviceLuidSupported = false;
+    if (hDevice->checkDeviceExtensions({"cl_khr_device_uuid"},
+                                       isKhrDeviceLuidSupported) !=
+            UR_RESULT_SUCCESS ||
+        !isKhrDeviceLuidSupported) {
+      return UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION;
+    }
+
+    cl_int *nodeMask = nullptr;
+
+    CL_RETURN_ON_FAILURE(clGetDeviceInfo(hDevice->CLDevice,
+                                         CL_DEVICE_NODE_MASK_KHR,
+                                         sizeof(cl_int), nodeMask, nullptr));
+
+    if (nodeMask == nullptr) {
+      return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+    }
+
+    return ReturnValue(*nodeMask);
```

Comment:
The node mask query returns the node mask directly, not a pointer to the node mask, so I think this should be:

```suggestion
    cl_uint nodeMask = 0;

    CL_RETURN_ON_FAILURE(clGetDeviceInfo(hDevice->CLDevice,
                                         CL_DEVICE_NODE_MASK_KHR,
                                         sizeof(cl_int), &nodeMask, nullptr));

    return ReturnValue(nodeMask);
```

Reply:
Note: The OpenCL test should be updated similarly.

Reply:
Yeah you are right. I have added this and fixed this same issue elsewhere.